### PR TITLE
Issue #31 - fix file cleanup problem in delete app version

### PR DESCRIPTION
**Pull request by scorbo2**
This PR fixes a bug in ProjectManager.removeExtensionVersion(), where the code was incorrectly using the `getBasename()` utility method to compute the location of associated signature files. The `getBasename()` method was inappropriate for this purpose, as it removes leading path elements, which must remain intact. Added a new utility method `switchExtension()` which can be used to replace the extension at the end of a file path without stripping leading path elements. Added unit tests for the new utility method.

Unrelated cleanup changes in this PR:
- reduce log level of an annoying log message on project save
- slight code cleanup in VersionManifest card related to retrieving the list model

### CODE REVIEW

File: src/test/java/ca/corbett/packager/project/ProjectManagerTest.java [hunk 1]
```diff
@@ -165,6 +165,42 @@ public void validateExtInfo_withBlankTargetAppVersion_shouldThrow() {
...
+    public void switchExtension_withMultipleExtensions_shouldReplaceLastExtension() throws Exception {
+        // GIVEN a file path with multiple extensions:
+        final String filePath = "extensions/My.Extension.Name-1.0.0.tar.gz";
+
+        // WHEN switching the extension to "zip":
+        final String newFilePath = projectManager.switchExtension(filePath, "zip");
+
+        // THEN the new file path should have the "zip" extension:
+        assertEquals("extensions/My.Extension.Name-1.0.0.tar.zip", newFilePath);
+    }
```
[the remark this answers is not in the document]
Comment (scorbo2) [reply src/test/java/ca/corbett/packager/project/ProjectManagerTest.java]: Wow, good catch. Fixed.

File: src/main/java/ca/corbett/packager/project/ProjectManager.java [hunk 2]
```diff
@@ -609,6 +612,26 @@ public static String getBasename(String filename) {
...
+        int index = filename.lastIndexOf(".");
+        String withoutExtension;
+        if (index == -1) {
+            withoutExtension = filename;
+        }
+        else {
+            withoutExtension = filename.substring(0, index);
+        }
+        return withoutExtension + "." + newExtension;
+    }
```
[the remark this answers is not in the document]
Comment (scorbo2) [reply src/main/java/ca/corbett/packager/project/ProjectManager.java]: Fixed and added tests for the new edge case.

File: src/main/java/ca/corbett/packager/ui/VersionManifestCard.java [hunk 3]
```diff
@@ -163,9 +163,8 @@ private void deleteApplicationVersion() {
             return;
         }
 
-        DefaultListModel<VersionManifest.ApplicationVersion> listModel = (DefaultListModel<VersionManifest.ApplicationVersion>)appVersionListField.getListModel();
-        VersionManifest.ApplicationVersion appVersion = listModel.getElementAt(selectedIndex);
-        listModel.removeElementAt(selectedIndex);
+        VersionManifest.ApplicationVersion appVersion = appVersionListField.getListModel().getElementAt(selectedIndex);
+        appVersionListField.getListModel().removeElementAt(selectedIndex);
```
[the remark this answers is not in the document]
Comment (scorbo2) [reply src/main/java/ca/corbett/packager/ui/VersionManifestCard.java]: The other code is wrong, though - swing-extras had a change in the last release that means we don't have to do that ridiculous cast anymore. I want to go through this code and remove all the unnecessary casts, but eh, not for this ticket.
